src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -844,11 +860,140 @@ private static string MakeShortCategoryString(string fullCategoryName)
+        /// Used during library search key navigation. Counts next selected member index.
+        /// </summary>
+        public void MoveSelection(NavigationDirection direction)
+        {
+            // We should have at least one category, otherwise we can't move.
+            if (!SearchRootCategories.Any())
+                return;
+
+            var offset = direction == NavigationDirection.Backward ? -1 : 1;
+
+            var selectedCategory = SearchRootCategories[selectedCategoryIndex];
+            var selectedMemberGroup = selectedCategory.MemberGroups.ElementAt(selectedMemberGroupIndex);
+
+            // If it's not the first/last member, we can move further.
+            if ((selectedMemberIndex != selectedMemberGroup.Members.Count() - 1 || direction != NavigationDirection.Forward) &&
+                (selectedMemberIndex != 0 || direction != NavigationDirection.Backward))
+            {
+                UnSelectOldMember();
+                selectedMemberIndex += offset;
+                SelectNewMember();
+                return;
+            }
+
+            // If it's first/last member, we should jump to next member group.
+            if ((selectedMemberGroupIndex != selectedCategory.MemberGroups.Count() - 1 || direction != NavigationDirection.Forward) &&
+                (selectedMemberGroupIndex != 0 || direction != NavigationDirection.Backward))
+            {
+                UnSelectOldMember();
+                selectedMemberGroupIndex += offset;
+                if (direction == NavigationDirection.Forward)
+                    selectedMemberIndex = 0;
+                else
+                    // Select the last member.
+                    selectedMemberIndex = selectedCategory.MemberGroups.ElementAt(selectedMemberGroupIndex).Members.Count() - 1;
+                SelectNewMember();
+                return;
+            }
+
+            // If it's first/last member group, we should jump to next category.
+            if ((selectedCategoryIndex != SearchRootCategories.Count() - 1 || direction != NavigationDirection.Forward) &&
+                (selectedCategoryIndex != 0 || direction != NavigationDirection.Backward))
+            {
+                UnSelectOldMember();
+                selectedCategoryIndex += offset;
+
+                if (direction == NavigationDirection.Forward)
+                {
+                    selectedMemberGroupIndex = 0;
+                    selectedMemberIndex = 0;
+                }
+                else
+                {
+                    var memberGroups = SearchRootCategories[selectedCategoryIndex].MemberGroups;
+                    selectedMemberGroupIndex = memberGroups.Count() - 1;
+                    // Select last member in member group. We are moving from bottom to top.
+                    selectedMemberIndex = memberGroups.ElementAt(selectedMemberGroupIndex).Members.Count() - 1;
+                }
+                SelectNewMember();
+            }
+        }
```

Comment:
Hmm... almost, but not quite, Anna. Sorry! `MoveSelection` method above does not need to be that complicated, let's make it absolutely very simple :) Okay to make your life easier, let me suggest something like this with more details:

```csharp
class SelectionNavigator
{
    internal enum NavigationDirection { Backward, Forward }
    
    // No default selection by default.
    private int selectedCategoryIndex = -1;
    private int selectedMemberGroupIndex = -1;
    private int selectedMemberIndex = -1;
    private NodeSearchElementViewModel selection = null;
    
    public NodeSearchElementViewModel CurrentlySelection
    {
        get { return selection; }
    }
    
    internal SelectionNavigator(ObservableCollection<SearchCategory> root)
    {
    }
    
    internal void MoveSelection(NavigationDirection direction)
    {
        var offset = direction == NavigationDirection.Backward ? -1 : 1;
        
        // First we update the selection index.
        selectedMemberIndex += offset;
        
        var selectedCategory = root[selectedCategoryIndex];
        var selectedMemberGroup = selectedCategory.MemberGroups.ElementAt(selectedMemberGroupIndex);

        // Clear the current selection, no matter what.
        selection.IsSelected = false;
        selection = null;
        
        if (selectedMemberIndex < 0)
        {
            // Handle underflow scenarios and update the following members...
            // 
            selectedCategoryIndex = ...;
            selectedMemberGroupIndex = ...;
            selectedMemberIndex = ...;
        }
        else
        {
            // Determine the current group size...
            var members = selectedMemberGroup.Members.Count();
            if (selectedMemberIndex >= members)
            {
                // Handle overflow scenarios and update the following members...
                // 
                selectedCategoryIndex = ...;
                selectedMemberGroupIndex = ...;
                selectedMemberIndex = ...;
            }
        }
        
        // Get the new selection and mark it as selected.
        selection = GetSelectionFromIndices();
        selected.IsSelected = true;
    }
    
    private NodeSearchElementViewModel GetSelectionFromIndices()
    {
        // Get the view model from indices like selectedMemberIndex...
    }
}

public partial class SearchViewModel : NotificationObject
{
    private SelectionNavigator selectionNavigator = null;
    
    public void MoveSelection(SelectionNavigator.NavigationDirection direction)
    {
        selectionNavigator.MoveSelection(direction);
    }
}
```